### PR TITLE
Added 'getApplicationSignature' to assist Android users generating the key hash

### PR DESCRIPTION
**Pull request by EddyVerbruggen**
Hi,

This should be easier than using those nasty OpenSSL commands to generate the hash keys.

Best,
Eddy

### CODE REVIEW

File: facebookConnectPlugin.js [hunk 1]
```diff
@@ -219,6 +234,10 @@ if (cordova.platformId == "browser") {
             exec(s, f, "FacebookConnectPlugin", "logout", []);
         },
 
+      getApplicationSignature: function (s, f) {
```
Comment (aogilvie) [inline facebookConnectPlugin.js]: The indentation on method names seems to be slightly off.

File: platforms/android/README.md [hunk 2]
```diff
@@ -4,6 +4,9 @@ To use this plugin you will need to make sure you've registered your Facebook ap
 
 If you plan on rolling this out on Android, please note that you will need to [generate a hash of your Android key(s) and submit those to the Developers page on Facebook](https://developers.facebook.com/docs/android/getting-started/facebook-sdk-for-android/) to get it working. Furthermore, if you are generating this hash on Windows (specifically 64 bit versions), please use version 0.9.8e or 0.9.8d of [OpenSSL for Windows](http://code.google.com/p/openssl-for-windows/downloads/list) and *not* 0.9.8k. Big ups to [fernandomatos](http://github.com/fernandomatos) for pointing this out!
 
+As a convenience you can use the plugin function `facebookConnectPlugin.getApplicationSignature()` to get the hash key.
+See [this document](http://plugins.telerik.com/plugin/facebook) on how to use this method.
```
Comment (aogilvie) [inline platforms/android/README.md]: While I think the documentation at the link provided is good, it should be put in here for completeness.

**Comment by aogilvie** [comment]
Can this be PRed to develop branch please?

File: platforms/android/src/org/apache/cordova/facebook/ConnectPlugin.java [hunk 3]
```diff
@@ -299,6 +301,21 @@ public void onCompleted(GraphUser user, Response response) {
                 }
             }
             return true;
+        } else if (action.equals("getApplicationSignature")) {
+            String sig = Settings.getApplicationSignature(webView.getContext());
+            if (sig == null) {
+              callbackContext.error("Could not determine signature.");
+            } else {
+              // strip the linefeed..
+              sig = sig.replace(System.lineSeparator(), "");
```
Comment (aogilvie) [inline platforms/android/src/org/apache/cordova/facebook/ConnectPlugin.java]: I think your Java indents are 2 spaces. Can you match the existing style? Thanks.

File: facebookConnectPlugin.js (hunk 1 above)
Comment (EddyVerbruggen) [inline facebookConnectPlugin.js]: You have very good eyes ;)

File: platforms/android/src/org/apache/cordova/facebook/ConnectPlugin.java (hunk 3 above)
Comment (EddyVerbruggen) [inline platforms/android/src/org/apache/cordova/facebook/ConnectPlugin.java]: Sure thing

**Comment by EddyVerbruggen** [comment]
All changes have been pushed to #980, closing this one.

**Pull request closed without merging**